calcrule_unconditional_cash_payment/calculation_rule.py

```diff
@@ -0,0 +1,128 @@
+    def active_for_object(cls, instance, context, type="account_payable", sub_type="cash_payment"):
+        return instance.__class__.__name__ == "Policy" \
+               and context in ["policyCreated"] \
+               and cls.check_calculation(instance)
+
+    @classmethod
+    def check_calculation(cls, instance):
+        class_name = instance.__class__.__name__
+        match = False
+        if class_name == "ABCMeta":
+            match = str(cls.uuid) == str(instance.uuid)
+        elif class_name == "PaymentPlan":
+            match = str(cls.uuid) == str(instance.calculation)
+        elif class_name == "Policy":
+            match = cls.check_calculation(instance.product)
+        elif class_name == "Product":
+            # if product → paymentPlans
+            payment_plans = PaymentPlan.objects.filter(benefit_plan=instance, is_deleted=False)
+            for pp in payment_plans:
+                if cls.check_calculation(pp):
+                    match = True
+                    break
+        return match
+
+    @classmethod
+    def calculate(cls, instance, **kwargs):
+        context = kwargs.get('context', None)
+        user = kwargs.get('user', None)
+        class_name = instance.__class__.__name__
+        if class_name == "Policy":
+            product = instance.product
+            payment_plan = PaymentPlan.objects.filter(benefit_plan=product, calculation=cls.uuid, is_deleted=False)
+            if payment_plan:
+                payment_plan = payment_plan.first()
+                cls.run_convert(instance=instance, context=context, convert_to='Bill', payment_plan=payment_plan,
+                                user=user)
+            return "conversion finished " + cls.calculation_rule_name
+
+    @classmethod
+    def get_linked_class(cls, sender, class_name, **kwargs):
+        list_class = []
+        if class_name is not None:
+            model_class = ContentType.objects.filter(model=class_name).first()
+            if model_class:
+                model_class = model_class.model_class()
+                list_class = list_class + [f.remote_field.model.__name__ for f in model_class._meta.fields
+                                           if f.get_internal_type() == 'ForeignKey'
+                                           and f.remote_field.model.__name__ != "User"]
+        else:
+            list_class.append("Calculation")
+        # because we have calculation in PaymentPlan
+        #  as uuid - we have to consider this case
+        if class_name == "PaymentPlan":
+            list_class.append("Calculation")
+        return list_class
+
+    @classmethod
+    def convert(cls, instance, convert_to, **kwargs):
+        context = kwargs.get('context', None)
+        payment_plan = kwargs.get('payment_plan')
+        convert_results = {}
+        if context == "policyCreated":
+            lumpsum_to_be_paid = None
+            invoice_label = None
+            if 'calculation_rule' in payment_plan.json_ext:
```

Comment:
@wzglinieckisoldevelo Please move 'obtaining values from json_ext' logic into separate private method. You can also as you wish move this into 'utils'.